build/Build.Steps.cs

```diff
@@ -134,11 +150,26 @@ partial class Build
 
             foreach (var app in testApps)
             {
-                DotNetBuild(x => x
-                    .SetProjectFile(app)
-                    .SetConfiguration(BuildConfiguration)
-                    .SetPlatform(Platform)
-                    .SetNoRestore(true));
+                if (TestPackageVersions.Versions.TryGetValue(app.Name, out var testedPackageVersions))
```

Comment:
If feels like this should be generated during the build of our `_build.csproj` since we are using the compiled version not the one generated during the workflow itself. It may make sense overall, but, at first it seems strange.

Reply:
I am not sure if I understand this comment correctly but I have added generation tool to the build pipeline to ensure that there is no missing generated files in the repository.

Reply:
ci will detect if the file changed so we are covered, but, `dotnet nuke Workflow` will run with the source file that was checked in, i.e.: before the targeting generating/updating the source was run. It should work, it is just somehow unexpected (at least to me).

Reply:
Well we can rearrange pipeline a bit, but it looks like a bigger rework.
Restore, build and execute generator before restore/build tests.
It is doable, if you are fine, I would create separate issue.